pallets/identity/src/lib.rs

```diff
@@ -203,14 +200,11 @@ decl_module! {
         // TODO: Remove this function. cdd_register_did should be used instead.
         pub fn register_did(origin, signing_items: Vec<SigningItem>) -> DispatchResult {
             let sender = ensure_signed(origin)?;
-            // TODO: Subtract proper fee.
-            let _imbalance = <T::Balances>::withdraw(
-                &sender,
-                Self::did_creation_fee(),
-                WithdrawReason::Fee.into(),
-                ExistenceRequirement::KeepAlive,
+            let sender_key = AccountKey::try_from(sender.encode())?;
+            T::ProtocolFee::charge_fee(
```

Comment:
`_register_did` function that is called after charging fee does additional checks and can return an error without making state changes.